code/modules/projectiles/guns/projectile/shotgun.dm

```diff
@@ -80,16 +80,82 @@
 	desc = "A sturdy shotgun with a longer magazine and a fixed tactical stock designed for non-lethal riot control."
 	icon_state = "riotshotgun"
 	mag_type = /obj/item/ammo_box/magazine/internal/shot/riot
-	sawn_desc = "Come with me if you want to live."
+	var/style = 1
 
 /obj/item/weapon/gun/projectile/shotgun/riot/attackby(obj/item/A, mob/user, params)
 	..()
+	var/temp_sawing = 0
 	if(istype(A, /obj/item/weapon/circular_saw) || istype(A, /obj/item/weapon/gun/energy/plasmacutter))
-		sawoff(user)
+		temp_sawing = 1
 	if(istype(A, /obj/item/weapon/melee/energy))
 		var/obj/item/weapon/melee/energy/W = A
 		if(W.active)
-			sawoff(user)
+			temp_sawing = 1
+	if(temp_sawing)

```

Comment:
You should definitely put it in it's own proc

Reply:
Ok, I put them into the old sawn off proc, even if I had to remake it fully